### PR TITLE
Adding BatchNorm

### PR DESCRIPTION
**Pull request by egretholier**
Cons:

- Yet another argument to AttentionLayers
- This is different from FFNBN in the [Leveraging Batch Normalization for Vision Transformers](https://openaccess.thecvf.com/content/ICCV2021W/NeurArch/papers/Yao_Leveraging_Batch_Normalization_for_Vision_Transformers_ICCVW_2021_paper.pdf) paper.

Pros:
- Time series transformer papers are empirically claiming better performance. [A Transformer-based Framework for Multivariate Time Series Representation Learning](https://arxiv.org/abs/2010.02803) made the original claim and [A Time Series is Worth 64 Words: Long-term Forecasting with Transformers](https://arxiv.org/abs/2211.14730) made it popular.

### CODE REVIEW

**Comment by lucidrains** [comment]
@egretholier hey, thanks for adding this! could you throw in a test as well as make sure it asserts out if autoregressive or has any masking (masked batch norm is doable, but probably too complex for what you wish to do)

**Comment by lucidrains** [comment]
@egretholier for the record, i did try https://arxiv.org/abs/2003.07845 back in the day (collaborating with Aran Komat during a time when transformers were relatively unknown) and did not see much of anything.
